llama_stack/testing/inference_recorder.py

```diff
@@ -248,6 +267,20 @@ async def _patched_inference_method(original_method, self, client_type, endpoint
         recording = _current_storage.find_recording(request_hash)
         if recording:
             response_body = recording["response"]["body"]
+            if (
+                isinstance(response_body, list)
+                and len(response_body) > 0
+                and isinstance(response_body[0], ChatCompletionChunk)
+            ):
+                # We can't replay chatcompletions with the same id and we store them in a sqlite database with a unique constraint on the id.
```

Comment:
can you explain this situation in more detail? does this happen because we have both ollama and vllm in the same DB or some other reason?

Reply:
@derekhiggins a bit confused -- this code is being changed during replay time, not recording time. The duplicate error would have happened during recording time right? How does this fix prevent that?

Reply:
It all boils down to
> we can't replay with the same id twice as LLS stores them with the ID as a primary key (the ollama provider doesn't hit this because it creates a new id for each response it gets from ollama)

Reply:
Ah this is an interaction of test replay with "inference store" -- which has nothing to do with the test infrastructure by itself. cc @ehhuang

Reply:
I think we should simply make inference_store robust to collisions with a `ON DUPLICATE IGNORE` kind of clause.

Reply:
Yeah that's a reasonable solution @ehhuang

Reply:
@derekhiggins that seems OK to me. can you note in the comment here we need this for `inference_store`?